simulators/gaia/__init__.py

```diff
@@ -151,104 +157,107 @@ def _execute(self, msg):
             if args[1] not in range(1024):
                 return self._error(1010)
         cmd = getattr(self, cmd)
-        return cmd(args)
+        raw_response = cmd(args)
+        tail = ' %s%s' % (self.cmd_id, self.tail)
+        return self.header + str(raw_response) + tail
 
     def _error(self, error_code):
         error_string = self.errors.get(error_code)
-        return 'ERROR(%d)[%s](%s)%c' % (
+        return '%sERROR(%d)[%s](%s) %s%c' % (
+            self.header,
             error_code,
             error_string,
             error_string.encode('hex'),
+            self.cmd_id,
             self.tail
         )
 
-    def _idn(self, _):
```

Comment:
Qua era giusto l'underscore, ho messo args perché in un primo momento volevo per recuperare l'id del comando da args, ma poi ho fatto diversamente pre cui va bene _. Lo correggo al termine della revisione (non ricordo più nemmeno come si fa... :-/ )